src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -0,0 +1,128 @@
+                        <View style={[styles.p1, styles.pl2]}>
+                            <Icon
+                                src={Expensicons.ArrowRight}
+                                fill={theme.icon}
+                            />
+                        </View>
+                    </View>
+                ),
+            })),
+        [policyCategories, selectedCategories, styles.alignSelfCenter, styles.disabledText, styles.flexRow, styles.p1, styles.pl2, theme.icon, translate],
+    );
+
+    const toggleCategory = (category: PolicyForList) => {
+        setSelectedCategories((prev) => ({
+            ...prev,
+            [category.value]: !prev[category.value],
+        }));
+    };
+
+    const toggleAllCategories = () => {
+        const isAllSelected = categoryList.every((category) => !!selectedCategories[category.value]);
+        setSelectedCategories(isAllSelected ? {} : Object.fromEntries(categoryList.map((item) => [item.value, true])));
+    };
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            style={[styles.defaultModalContainer]}
+            testID={WorkspaceCategoriesPage.displayName}
+            shouldShowOfflineIndicatorInWideScreen
+        >
+            <HeaderWithBackButton
+                icon={Illustrations.FolderOpen}
+                title={translate('workspace.common.categories')}
+                shouldShowBackButton={isSmallScreenWidth}
+            />
+            <View style={[styles.ph5, styles.pb5]}>
+                <Text style={[styles.textNormal, styles.colorMuted]}>{translate('workspace.categories.subtitle')}</Text>
+            </View>
+            {categoryList.length ? (
+                <SelectionList
+                    canSelectMultiple
+                    sections={[{data: categoryList, indexOffset: 0, isDisabled: false}]}
+                    onSelectRow={toggleCategory}
+                    onSelectAll={toggleAllCategories}
+                    showScrollIndicator
+                />
+            ) : (
+                <WorkspaceEmptyStateSection
+                    title={translate('workspace.categories.emptyCategories.title')}
+                    icon={Illustrations.EmptyStateExpenses}
+                    subtitle={translate('workspace.categories.emptyCategories.subtitle')}
+                />
+            )}
+        </ScreenWrapper>
+    );
+}
+
+WorkspaceCategoriesPage.displayName = 'WorkspaceCategoriesPage';
+
+export default withPaidPolicyAccessOrNotFound()(
+    withAdminPolicyAccessOrNotFound()(
+        withOnyx<WorkspaceCategoriesPageProps, WorkspaceCategoriesOnyxProps>({
+            policyCategories: {
+                key: ({route}) => `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${route.params.policyID}`,
+            },
+        })(WorkspaceCategoriesPage),
+    ),
+);
```

Comment:
I think you can try:

```suggestion
export default compose(
    withPaidPolicyAccessOrNotFound,
    withAdminPolicyAccessOrNotFound,
    withOnyx<WorkspaceCategoriesPageProps, WorkspaceCategoriesOnyxProps>({
        policyCategories: {
            key: ({route}) => `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${route.params.policyID}`,
        },
    }),
)(WorkspaceCategoriesPage);
```

You'll also have to import compose

Reply:
Hmm nvm, I think that crashes the App 😅